cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -935,6 +1152,21 @@ struct JSONToStackOp {
   }
 };
 
+/**
+ * @brief Function object used to filter for brackets and braces that represent push and pop
+ * operations
+ */
+struct JSONWithRecoveryToStackOp {
+  template <typename StackSymbolT>
+  constexpr CUDF_HOST_DEVICE fst::stack_op_type operator()(StackSymbolT const& stack_symbol) const
+  {
+    return (stack_symbol == '{' || stack_symbol == '[')   ? fst::stack_op_type::PUSH
+           : (stack_symbol == '}' || stack_symbol == ']') ? fst::stack_op_type::POP
+           : (stack_symbol == '\n')                       ? fst::stack_op_type::RESET
+                                                          : fst::stack_op_type::READ;
```

Comment:
Could we use a switch here? I find this less readable, but I understand it might just be a personal preference.

Reply:
Done 👍